api/src/pcapi/routes/backoffice/accounts/blueprint.py

```diff
@@ -908,12 +901,21 @@ def review_public_account(user_id: int) -> utils.BackofficeResponse:
                 return redirect(get_public_account_link(user_id), code=303)
 
     try:
+        if eligibility == users_models.EligibilityType.AGE18 and user.has_underage_beneficiary_role:
```

Comment:
(question) Pourquoi rajouter ce check et ce changement d'expiration sur le deposit ?

Il me semble que la gestion de l'expiration des deposit existe déjà.
Et sinon, je ne suis pas pour que l'ajout d'une règle se fasse dans un blueprint du backoffice.

Reply:
Surtout, je ne pense pas qu'on octroie un crédit 18 à un jeune qui n'a pas encore atteint 18 ans. Donc son crédit "underage" est forcément déjà expiré (au jour de ses 18 ans), non ?
@thconte ou @lixxday vous êtes sûrement plus pointus sur le sujet

Reply:
Si il y a une fonction pour le faire dans le backend je ne l'ai pas trouvée. Celle qui est appelée vérifie bien qu'il n'y ai pas de déposit valide pour les bénéficiaires de moins d e18 ans.

Reply:
j'ai oublié de répondre a la deuxieme partie de la question: la regle rajouté ne conerne que l'activation manuelle part la fraude jeune c'est pour ça que je ne l'ai pas mise dans api. Sachant qu'il était impossible de donner un credit 18 a un underage depuis au moins 6 mois c'est que ce cas est tres a la marge.

Reply:
Effectivement, d'une part lorsqu'un utilisateur a 18 ans, son crédit underage est censé être expiré (par construction à sa création, cf `compute_underage_deposit_expiration_datetime`), d'autre part si on lui octroie un nouveau deposit (celui de ses 18 ans), c'est celui qui sera utilisé même si l'autre n'est pas expiré (cf la property `deposit` du model `User`).
Aussi la fonction appelée pour une revue manuelle c'est `handle_ok_manual_review`.

Reply:
De ce que j'ai compris, le cas qu'on veut gérer ici c'est : Un jeune a un deposit underage actif et on veut lui octroyer un deposit 18 ans.
Je vois pas comment on peut en arriver là mais je suppose que le cas est apparu au moins une fois si on a dû rajouter une vérification.
Après discussion je suis ok pour laisser la query d'expiration du deposit ici.
Du coup je pense aussi qu'il faudrait remplacer dans la condition
```suggestion
        if eligibility == users_models.EligibilityType.AGE18 and user.has_active_deposit:
```

Reply:
Comme la requete ne modifie que les deposit actif, cette verification ne sert a rien. Si l'utilisateur n'a pas de deposit actif alors la requet ne fera rien ;)

Reply:
Dur vendredi